src/ReflectionClosure.php

```diff
@@ -271,6 +272,9 @@ public function getCode()
                         case T_CURLY_OPEN:
                         case T_DOLLAR_OPEN_CURLY_BRACES:
                         case '{':
+                            if ($context === 'string' && $token[0] !== '{') {
```

Comment:
This context switching should happen only when T_DOLLAR_OPEN_CURLY_BRACES.
Something like this:
```php
case T_DOLLAR_OPEN_CURLY_BRACES:
  if ($context === 'string') {
    $context = 'root';
  }
 // no break
case '{':
  $code .= ...
```

Reply:
Yes, but also T_CURLY_OPEN because:
- after remove: T_CURLY_OPEN
```php
$c = function () {
    $user = ['username' => 'abdzakoxa'];
    return "{$user['username']}";
};
$u = $this->s($c);
```
- test result:
ParseError : syntax error, unexpected end of file
 closure://function () {
             $user = ['username' => 'abdzakoxa'];
             return "{$user['username']}:4